packages/design-mui/src/components/MuiFilters/hooks/useMuiFilters.tsx

```diff
@@ -0,0 +1,64 @@
+import type { XCollectionProps } from "@hatchifyjs/react-ui"
+import { capitalize } from "lodash"
+import { useEffect, useRef, useState } from "react"
+
+type HatchifyMuiFilters = {
+  fields: string[]
+}
+
+export default function useMuiFilter(
+  allSchemas: XCollectionProps["allSchemas"],
+  schemaName: XCollectionProps["schemaName"],
+  include: string[],
+): HatchifyMuiFilters {
+  const allSchemasRef = useRef(allSchemas)
+  const includeRef = useRef(include)
+  const defaultFields = Object.entries(
+    allSchemasRef.current[schemaName].attributes,
+  )
+    .filter(([, attr]) =>
+      typeof attr === "object"
+        ? attr.type === "string" || attr.type === "date" || attr.type === "enum"
+        : attr === "string" || attr === "date",
```

Comment:
We previously had another check for `attr === "enum"`, but that should never exist since enum type will always have to be accompanied by an array of values.